pyomo/contrib/trustregion/PyomoInterface.py

```diff
@@ -48,12 +60,15 @@ def visit(self, node, values):
         # things later; i.e., accessing the private _fcn attribute
         # below).
         for arg in list(values)[1:]:
```

Comment:
See the note above, but if you start treating nonpyomo_leaf_nodes as fixed arguments and ignoring them from the TR models, then we can remove the `[1:]` restriction here, an probably relax the set of allowable ExternalFunction types to "all".